geotrek/api/v2/filters.py

```diff
@@ -695,6 +694,11 @@ def get_schema_fields(self, view):
 
 class GeotrekSiteFilter(BaseFilterBackend):
     def filter_queryset(self, request, queryset, view):
+        root_sites_only = request.GET.get('root_sites_only')
+        if root_sites_only:
+            for site in queryset:
```

Comment:
D'un point de vue performances c'est pas terrible d'itérer sur tous les éléments d'une queryset. Les fonctions filter / exclude sont plus performantes, il faudrait essayer de privilégier ça.
Je ne sait pas à quel point c'est faisable facilement dans ton cas mais peut-être qu'en annotant la queryset au préalable (dans ton viewset), ca permettrai de filtrer directement toute la qs plutôt que d'itérer dessus.

Reply:
Je remplace par `queryset = queryset.filter(parent=None)`
d'après [l'implémentation de is_root_node()](https://github.com/django-mptt/django-mptt/blob/551d266e464d1e54c28eb8244e271bcbde9daea5/mptt/models.py#L797) qui vérifie juste si l'objet a un parent